lib/ansible/playbook/included_file.py

```diff
@@ -184,7 +184,7 @@ def process_include_results(results, iterator, loader, variable_manager):
                         for from_arg in new_task.FROM_ARGS:
                             if from_arg in include_args:
                                 from_key = from_arg.replace('_from', '')
-                                new_task._from_files[from_key] = templar.template(include_args.pop(from_arg))
+                                new_task._from_files[from_key] = templar.template(include_args.get(from_arg))
```

Comment:
this does not seem like the correct change, instead we need to make sure that the vars_from are loaded before pasisng to param hashing

Reply:
@bcoca those are 2 different things.  `role` deduplication, vs `IncludedFile` deduplication.

We "dedupe" `IncludedFile` due to many hosts including the same file, so we squash them into 1 object.  So it's a different mechanism.

And the means in which we dedupe or squash `IncludedFile` depends on `args` on in this case `include_args`.  So if we `.pop` them, then those args cannot be used for deduping.

Consider the example of individual hosts using a different `vars_from`.  Right now we would squash all of those hosts onto a single `IncludedFile` that didn't account for `vars_from`.

This seems like the right change to me.

Reply:
but you dont have full args till you read the file. I becomes specially fun when the from itself is a template